### ISSUE TITLE
Saving configuration options no longer works reliably since version 16.2.0

### ISSUE THREAD

**Issue by Greg-21** [eranif/codelite#3041]
There are several configuration options in CodeLite 16.2.0 that do not remember their new state at all. No matter what I do, they do not want to be changed from enabled to disabled or vice versa. The only things that change are GUI widgets, but not for long, as after closing the window and reopening it, everything returns to the previous state.

This seems to be a recent regression, as configurations worked perfectly fine in CodeLite 16.1.0.

**Comment by eranif** [comment]
can you elaborate on which settings are not being saved?
also, your os

**Comment by Greg-21** [comment]
After a brief investigation, the following configuration options do not seem to work as expected:
* "Guides" tab:
   * Enable background color (the one related to the debugger line)
* "Edit" tab:
   * Smart curly brackets
   * Smart square brackets and parentheses
   * Smart quotes
   * Wrap with quotes
   * Wrap with brackets
   * Enable zoom with mouse scroll
* "Caret / Scrolling" tab:
   * Use block caret
   * Allow caret to be placed beyond the end of line
* "Code" tab:
   * Control key
   * Alt key
   * Class template
* "Folding" tab:
   * Highlight active fold block
* "Tabs" tab:
   * Show close button on tabs
   * Show file path on tab label (maybe this is what was reported by other user in #3038)

I'm not sure if these are all the options that do not work correctly. There may be more of them, but I haven't come across them yet. This is on Windows.

**Comment by eranif** [comment]
Windows have various flavors, are you using WIndows 11 by any chance?
Anyways, see my fix here:

https://github.com/eranif/codelite/commit/2da8e2ae9a01d76afd46086f956e6ec4402f8c9a

**Comment by Greg-21** [comment]
I've just installed the previous version of CodeLite, 16.1.0, and it looks like the configuration options I listed above in my previous comment did not work there either. Now I'm confused, as I definitely changed these options. Most likely, I just didn't notice that some of my tweaks were not saved back then.

Speaking of the operating system, I have been using Windows 8.1 for years, and nothing has changed in this regard recently. With that in mind, it looks like this issue is not unique to Windows 11.
